azext_iot/digitaltwins/providers/model.py

```diff
@@ -7,13 +7,45 @@
 import json
 from azext_iot.common.utility import process_json_arg, scantree, unpack_msrest_error
 from azext_iot.digitaltwins.providers.base import DigitalTwinsProvider
-from azext_iot.digitaltwins.providers import ErrorResponseException
+from azext_iot.sdk.digitaltwins.dataplane.models import ErrorResponseException
 from knack.log import get_logger
 from knack.util import CLIError
 
 logger = get_logger(__name__)
 
 
+def get_model_dependencies(model):
+    """Return a list of dependency DTMIs for a given model"""
+    dependencies = []
+
+    if "contents" in model:
+        components = [item["schema"] for item in model["contents"] if item["@type"] == "Component"]
```

Comment:
We talked about some additional updates for the component schema value processing (can be an instance of `str` or `dict`, handling is like we have done in extends ) + some unit tests to exercise

Reply:
This is a good reference https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelQuery.cs#L130